heron/schedulers/src/java/com/twitter/heron/scheduler/aurora/AuroraCLIController.java

```diff
@@ -111,29 +113,43 @@ public void removeContainers(Set<PackingPlan.ContainerPlan> containersToRemove)
   }
 
   @Override
-  public void addContainers(Integer count) {
+  public Set<Integer> addContainers(Integer count) {
     //aurora job add <cluster>/<role>/<env>/<name>/<instance_id> <count>
     //clone instance 0
     List<String> auroraCmd = new ArrayList<>(Arrays.asList(
-        "aurora", "job", "add", "--wait-until", "RUNNING", jobSpec + "/0", count.toString()));
-
-    if (isVerbose) {
-      auroraCmd.add("--verbose");
-    }
+        "aurora", "job", "add", "--wait-until", "RUNNING",
+        jobSpec + "/0", count.toString(), "--verbose"));
 
     LOG.info(String.format("Requesting %s new aurora containers %s", count, auroraCmd));
-    if (!runProcess(auroraCmd)) {
+    StringBuilder stderr = new StringBuilder();
+    if (!runProcess(auroraCmd, null, stderr)) {
       throw new RuntimeException("Failed to create " + count + " new aurora instances");
     }
+
+    if (stderr.length() <= 0) { // no container was added
+      LOG.info("empty output by Aurora");
+      return new HashSet<Integer>();
+    }
+    String stdoutStr = stderr.toString();
+    String pattern = "Querying instance statuses: [";
+    int idx1 = stdoutStr.indexOf(pattern) + pattern.length();
```

Comment:
Good to make this parsing logic as a method below

Reply:
updated